.github/workflows/ci-master.yml

```diff
@@ -0,0 +1,217 @@
+                  restore-keys: ${{ runner.os }}-composer-
+
+            - name: Install dependencies
+              run: composer install
+
+            - name: PHP Lint
+              run: composer config-eventespressocs && composer run-script lint:skip-warnings
+              # Allow failure for PHP > 7.3
+              continue-on-error: ${{ matrix.php-versions != '7.3' }}
+
+    php-unit-tests:
+        runs-on: ${{ matrix.operating-system }}
+        needs: changed
+        # Run only if PHP files have changed
+        if: ${{ needs.changed.outputs.php == 'true' }}
+        env:
+            PHPUNIT_VER: v7 # default
+            WP_VERSION: latest
+            WP_MULTISITE: 0
+        strategy:
+            matrix:
+                operating-system: [ubuntu-latest]
+                php-versions:
+                    ['5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1']
+        name: Unit Tests with PHP ${{ matrix.php-versions }}
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            - name: Set PHPUnit version
+              run: |
+                  if [[ -n "${{ matrix.php-versions }}" ]]; then
+                    case "${{ matrix.php-versions }}" in
+                      7.0)
+                        echo "Using PHPUnit 6"
+                        echo "PHPUNIT_VER=v6" >> $GITHUB_ENV
+                        ;;
+                      5.6)
+                        echo "Using PHPUnit 5"
+                        echo "PHPUNIT_VER=5.7.22" >> $GITHUB_ENV
+                        ;;
+                    esac
+                  fi
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-versions }}
+                  tools: phpunit:${{ env.PHPUNIT_VER }}
+                  extensions: mysql # needed for PHP < 7.1
+
+            - name: Start mysql service
+              run: sudo /etc/init.d/mysql start
+
+            - name: Get composer cache directory
+              id: composercache
+              run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            - name: Cache composer dependencies
+              uses: actions/cache@v2
+              with:
+                  path: ${{ steps.composercache.outputs.dir }}
+                  # Use composer.lock for key.
+                  key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+                  restore-keys: ${{ runner.os }}-composer-
+
+            - name: Check PHP Version
+              run: php -v
+
+            - name: Composer install
+              run: composer install --optimize-autoloader --prefer-dist
+
+            - name: Install WP Tests
+              run: bash tests/bin/install-wp-tests.sh wordpress_test root root localhost latest
+
+            - name: Run PHP Unit Tests
+              run: phpunit --configuration tests/phpunit.xml
```

Comment:
love how this is all laid out... much easier to follow than Travis, although there does appear to be some extra complexity